sdcm/commit_log_check_thread.py

```diff
@@ -0,0 +1,180 @@
+
+@dataclass
+class CommitlogConfigParams:
+    def __init__(self, db_cluster, ):
+        with db_cluster.cql_connection_patient(
+                node=db_cluster.nodes[0],
+                connect_timeout=300,) as session:
+            self.total_space_in_mb = int(session.execute(
+                "SELECT value FROM system.config WHERE name='commitlog_total_space_in_mb'").one().value)
+            self.use_hard_size_limit = bool(strtobool(session.execute(
+                "SELECT value FROM system.config WHERE name='commitlog_use_hard_size_limit'").one().value))
+            self.segment_size_in_mb = int(session.execute(
+                "SELECT value FROM system.config WHERE name='commitlog_segment_size_in_mb'").one().value)
+
+            if self.total_space_in_mb == -1:
+                self.total_space_in_mb = "scylla_memory_total_memory"
+            else:
+                self.total_space_in_mb = self.total_space_in_mb * 1048576
+
+    total_space_in_mb: int | str
+    use_hard_size_limit: bool
+    segment_size_in_mb: int
+
+
+@dataclass
+class PrometheusQueries:
+    def __init__(self, commitlog_params: CommitlogConfigParams):
+        self.overflow_commit_log_directory = self.overflow_commit_log_directory.format(
+            commitlog_total_space_in_mb=commitlog_params.total_space_in_mb,
+            commitlog_segment_size_in_mb=commitlog_params.segment_size_in_mb
+        )
+
+        self.zero_free_segments = self.zero_free_segments.format(
+            commitlog_total_space_in_mb=commitlog_params.total_space_in_mb,
+            commitlog_segment_size_in_mb=commitlog_params.segment_size_in_mb
+        )
+    # Prometheus queries with detailed description below
+    overflow_commit_log_directory = ("scylla_commitlog_disk_total_bytes>({commitlog_total_space_in_mb}"
+                                     "%2B({commitlog_segment_size_in_mb}%2A1048576))")
+    """
+        returns commitlog total size on disk if commit log directory exceed the limit
+
+        commit_log_directory_query description:
+        %2A is *
+        %2B is +
+
+        {commitlog_total_space_in_mb} :
+        if commitlog_total_space_in_mb is not -1 use  commitlog_total_space_in_mb*1048576
+        else commitlog_total_space_in_mb='scylla_memory_total_memory' metric
+        see CommitLogCheckThread.init_commitlog_params_from_db()
+
+        + {commitlog_segment_size_in_mb}*1048576) :
+        add size of 1 segment,  Scylla actually stops creating new segments not when the next segment
+        would cause us to exceed the value,
+        but instead it checks if we've already reached or exceeded the limit.
+
+        scylla_commitlog_disk_total_bytes > :
+        filter all good values when scylla_commitlog_disk_total_bytes<= commitlog_total_space_in_mb + 1 segment
+        and return from Prometheus only bad
+    """
+
+    zero_free_segments = \
+        ("scylla_commitlog_disk_active_bytes>((scylla_commitlog_disk_total_bytes>{commitlog_total_space_in_mb})"
```

Comment:
I'm not that familiar with prometheus queries format but for the `scylla_commitlog_disk_total_bytes>{commitlog_total_space_in_mb}` part we should have a `>=` (it doesn't matter now because in most default configurations `commitlog_total_space_in_mb` is not actually a multiple of segment size)

Reply:
fixed